### ISSUE TITLE
do not work on cocos2d-x 3.2

### ISSUE THREAD

**Issue by aidyelee** [raymondlu/super-animation-samples#34]
in cocos2d-x 3.2,  the node can't be seen.

**Comment by aidyelee** [comment]
anybody help please, the "super animation node " does not work on cocos2d-x 3.2, HELP!

**Comment by vkbsb** [comment]
Did you try calling playSection on the anim node? Have you used the super
anim node on prev versions of cocos2d-x?

**Comment by aidyelee** [comment]
Yeah, of course!  Both in cocos2d-x-2.2 and cocos2d-x-3.1,  they all work very well!

**Comment by aidyelee** [comment]
But in 3.2, it failed , no error exported, but the node is SuperAnimNode is invisible

**Comment by ghost** [comment]
I also have problem with this, do you using Lua or cpp ?  For me ( at cpp ) the animation is always covered by other sptites so if you create empty scene the animation will be visible. I dont know how to make it working correctly. Anyone can help?

**Comment by aidyelee** [comment]
Thank for your tips, I got it . Just optimize the "onDraw" in "SuperAnimNode"  will be fine.

**Comment by aidyelee** [comment]
redefine 
//virtual void draw(cocos2d::Renderer *renderer, const cocos2d::Mat4 &transform, bool flags); 
virtual void draw(cocos2d::Renderer *renderer, const cocos2d::Mat4 &transform, uint32_t flags);

implement it as follow:
_insideBounds = (flags & FLAGS_TRANSFORM_DIRTY) ? renderer->checkVisibility(transform, _contentSize) : _insideBounds;
    if(!_insideBounds){
        return;
    }

_customCommand.init(_globalZOrder);
_customCommand.func = CC_CALLBACK_0(SuperAnimNode::onDraw, this, transform, flags);
renderer->addCommand(&_customCommand);

OK DONE!

**Comment by bsandro** [comment]
Thanks aidyelee, your solution works perfectly; however now I have very strange rendering glitches - some frames in animation are stretched horizontally, vertically or even in both directions. Does anyone else has this problem or maybe even solution for it? Thanks a lot in advance.

**Comment by raymondlu** [comment]
I've just committed samples for cocos2d-3.2, please have a check.
thank you, @aidyelee

**Comment by bsandro** [comment]
@raymondlu  I tried this sample and textures were stretched as well :C
![2014-09-18 14 17 56](https://cloud.githubusercontent.com/assets/2354657/4318337/4b092a62-3f1d-11e4-84c0-ad6e3c987bc7.png)

**Comment by aidyelee** [comment]
@bsandro Me too, but only on Mac. You can try it on Phone , it works well.  
And the "reason of the  mess on Mac"  is out of my control ,  but to me , I don't care it too much ...

**Comment by bsandro** [comment]
@aidyelee that was actual iphone 5s screenshot.

**Comment by rekcah1986** [comment]
The mac simulator is useful for us, please help.^_^
@raymondlu

**Comment by raymondlu** [comment]
@bsandro is your game 64 bit or 32 bit?

**Comment by raymondlu** [comment]
@rekcah1986 how about your game? 64 bit or 32 bit?

**Comment by bsandro** [comment]
@raymondlu 64-bit.

**Comment by bsandro** [comment]
Wow that is really interesting - 32bit version (4S or android) works just fine, but 64bit is having issues I mentioned before.

**Comment by raymondlu** [comment]
there is an unknown bug for 64bit...
i will try to fix it soon, sorry to put you guys into trouble.

**Comment by aidyelee** [comment]
come on raymond, i 'll put my eye on you😊

**Comment by rekcah1986** [comment]
The bug only exist with cocos simulator on osx. Can we build it as 32bit to resolve it ?

**Comment by raymondlu** [comment]
@rekcah1986 you can have a try, build your simulator as 32bit version.

**Comment by rekcah1986** [comment]
@raymondlu I'm failed, 555555555555.

**Comment by raymondlu** [comment]
hey, guys, I have just fixed the 64bit bug for cocos2d-x-3.1 and cocos2d-x-3.2, please have update your code, any problem please let me know.

**Comment by aidyelee** [comment]
hey raymond, u r the man!!!

**Comment by rekcah1986** [comment]
Perfect~

**Comment by aidyelee** [comment]
Hi raymond, the SAC works very good now, but I found the function setAnchorPoint(0.5f, 0.5f) in "SuperAnimNode::Init" failed,  it  works as setAnchorPoint(1,1) always. (Both in iOS and Android)

But irregularly it works fine in Win32....  so I set it to be " setAnchorPoint(1,1)"  to make it work as one

**Comment by aidyelee** [comment]
Oh, it's my fault, forget it :)
